client/app/bundles/HelloWorld/components/ResturantItem.jsx

```diff
@@ -0,0 +1,99 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+export default class ResturantItem extends React.Component {
+
+    static propTypes = {
+        resturant: PropTypes.object.isRequired,
+        tenbis: PropTypes.string.isRequired
+    };
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            distance: 'Unknown',
+            distance_loaded: false
+        }
+    }
+
+    updateDistance() {
+        navigator.geolocation.getCurrentPosition((position) => {
```

Comment:
This kind of logic should be extracted to an external utility and handled via callbacks or promises